tests/unit/basic-authentication-interceptor.js

```diff
@@ -0,0 +1,106 @@
+describe('basic authentication interceptor', function() {
+	var $rootScope,
+		$q,
+		BasicValidationInterceptor,
+		deferred,
+		response;
+
+	function jsonHeaders () {
+		return {
+			'content-type': 'application/json'
+		};
+	}
+
+	function responseHandler () {
+		this.success = function() {};
+	}
+
+	function setupResponsePromise () {
+		deferred = $q.defer();
+		response = new responseHandler();
+
+		spyOn(response, 'success');
+
+		deferred
+			.promise
+			.then(response.success);
+	}
+
+	beforeEach(module('ucApp'));
+
+	beforeEach(function() {
+		inject(function(_$rootScope_, _$q_, _BasicValidationInterceptor_) {
+			$rootScope = _$rootScope_;
+			$q = _$q_;
+			BasicValidationInterceptor = _BasicValidationInterceptor_
+		});
+	});
+
+	it('is defined', function() {
+		expect(BasicValidationInterceptor).toBeDefined();
+	});
+
+	describe('a json response response', function() {
+		beforeEach(setupResponsePromise);
+
+		it('accepts an array', function() {
+			deferred
+				.resolve(BasicValidationInterceptor.response({
+					headers: jsonHeaders,
+					data: [
+						'foo',
+						'bar'
+					]
+				}));
+
+			$rootScope.$digest();
+				
+			expect(response.success).toHaveBeenCalled();
```

Comment:
One thing that comes to mind here is that this is asserting on `success` which is really an implementation detail. I wonder if this is being made trickier by the fact that the function being tested sometimes returns a (rejected) promise and sometimes a value. If it always returned a promise it might be easier to test (the docs suggest this is allowed). Along the lines of: 

```javascript
      BasicValidationInterceptor.response({
        headers: jsonHeaders,
        data: [
          'foo',
          'bar'
        ]
      })
      .then(function () {
        // passes
      }, function () {
        // fails the test
      })
```

Just a suggestion as I haven't tried testing interceptors before so I might be barking up the wrong tree.